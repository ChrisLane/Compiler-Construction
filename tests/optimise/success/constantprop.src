// This is to test constant propagation. Variables will be replaced by their values.
main() {
    var x = 50;
    x = (10+20) / 3;
    print $x;
}
