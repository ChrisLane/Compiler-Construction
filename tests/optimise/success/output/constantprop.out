
main (  ) { 
    New ( "x" = Const 10 ) In { 
        Print ( Const 10 ) 
    } 
}

